src/main/java/it/pagopa/ecommerce/commons/utils/NpgApiKeyHandler.java

```diff
@@ -0,0 +1,173 @@
+package it.pagopa.ecommerce.commons.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vavr.control.Either;
+import it.pagopa.ecommerce.commons.client.NpgClient;
+import it.pagopa.ecommerce.commons.exceptions.NpgApiKeyConfigurationException;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * This class handles all NPG api keys such as default one and per methods psp.
+ * Use {@link NpgApiKeyHandlerBuilder} class to initialize a new class instance
+ * api keys
+ */
+public class NpgApiKeyHandler {
+
+    /**
+     * NpgApiKeyHandler builder class
+     */
+    public static class NpgApiKeyHandlerBuilder {
+        private String defaultApiKey;
+        private final Map<NpgClient.PaymentMethod, NpgPspApiKeysConfig> methodsApiKeyMapping = new EnumMap<>(
+                NpgClient.PaymentMethod.class
+        );
+
+        /**
+         * Set the default NPG api key
+         *
+         * @param defaultApiKey the default api key to be set
+         * @return this builder instance
+         */
+        public NpgApiKeyHandlerBuilder setDefaultApiKey(String defaultApiKey) {
```

Comment:
[naming] Usually fluent interfaces use with instead of add in method names

```suggestion
        public NpgApiKeyHandlerBuilder withDefaultApiKey(String defaultApiKey) {
```

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/188/commits/17f5fabeeddc26d002749691174e3a9fde598cf2